src/coreclr/binder/assemblyname.cpp

```diff
@@ -51,38 +51,28 @@ namespace BINDER_SPACE
         DWORD dwRefOrDefFlags = 0;
         DWORD dwHashAlgId = 0;
 
-        if (fIsDefinition)
-        {
-            // Get the assembly token
-            IF_FAIL_GO(pIMetaDataAssemblyImport->GetAssemblyFromScope(&mda));
-        }
+        ReleaseHolder<IMDInternalImport> pIMetaDataAssemblyImport;
+
+        PEKIND PeKind = peNone;
+        DWORD dwPAFlags[2];
+        IF_FAIL_GO(BinderAcquireImport(pPEImage, &pIMetaDataAssemblyImport, dwPAFlags));
+        IF_FAIL_GO(AssemblyBinderCommon::TranslatePEToArchitectureType(dwPAFlags, &PeKind));
```

Comment:
nit: can we move `SetArchitecture` up (or move this down) so they are closer together?